ds_test.go

```diff
@@ -48,6 +49,23 @@ func newDS(t *testing.T) (*Datastore, func()) {
 	}
 }
 
+func newDSSync(t *testing.T, sync bool) (*Datastore, func()) {
+	path, err := ioutil.TempDir(os.TempDir(), "testing_badger_")
+	if err != nil {
+		t.Fatal(err)
+	}
+	opts := &DefaultOptions
+	opts.SyncWrites = sync
```

Comment:
Doesn't this change the default options for everything?

Reply:
Not sure what you mean?

Reply:
DefaultOptions is get by reference, then changed.

Reply:
Hmm I see, I don't think so winc this is instantiating a new default options instance and a reference to that yea?

Reply:
> DefaultOptions is get by reference, then changed.

Actually my previous comment is incorrect, you are quite correct and i have pushed a commit fixing this.